### ISSUE TITLE
prevent creation of data group if error is encountered upon registering records

### ISSUE THREAD

**Issue by kdionisio** [HumanExposure/factotum#659]
Upon attempting to create a new data group from here http://factotum.epa.gov/datasource/33/datagroup_new/, after hitting 'submit', an error was seen:

(1406, "Data too long for column 'url' at row 1")

Note the error was an 'external' error rather than an error message at the top of the Factotum application.

However, the application seems to still have created a data group even though the error was thrown. The data group only includes a subset of the records that were included in the register records csv, which user was attempting to register.

Definition of Done:
if an error is encountered when registering records/creating a data group, an error should be seen within the Factotum application and a data group should not be created

**Comment by kdionisio** [comment]
@debboutr   Would this be resolved as part of #910 or not?

**Comment by debboutr** [comment]
With a quick look, I would say no, I just tried to make a new datagroup on `dev` w/ an invalid URL and the error was thrown but it doesn't propogate out to the user to see, the datagroup never gets created, which is what this ticket adresses, but the URL error never makes it out to the user, this is the url in BOTH the DataGroup as well as the DataDocument, if there isn't a valid URL, say `http:www.123456789101112.com` in the case of the DataGroup, it won't be made but the runserver will err, and in the case of the DataDocument, it will be made, and an invalid URL in the datadocument field doesn't stop the document from being made. This is all going to change w/ @mcovalt 's refactor ticket, but we should make sure that it works as expected.

**Comment by kdionisio** [comment]
Update - as of 8/16, which is after pushing the data group page overhaul to production, this issue is still occurring

**Comment by debboutr** [comment]
@kdionisio  I'm going through the validation when registering records (i.e. creating a DataGroup). It appears that we don't make `GroupType` required for the DataGroup nor do we make the DocumentType required for the DataDocument. Do we want to have a `null` value for GroupType when we could have `Unidentified` as well? 
For the DocumentType then, we should allow for the `document_type` column of the CSV to be blank? or that it wouldn't be there at all?

**Comment by kdionisio** [comment]
@debboutr I believe we had stated previously that Group Type and Document Type could NOT be null. This is the reason we have the 'unidentified' assignment for document type, if you don't know what a document type is, you use 'unidentified' rather than null. So I would be OK with instituting this requirement if it is not already present in validation.

**Comment by debboutr** [comment]
@kdionisio sorry that this seems to be lingering on or so long, there is quite a bit of logic that we had been working off of with the old datagroup_create view. Now that I have made both `GroupType` and `DocumentType` required, I'm thinking that we will have to add the document type of "Unidentified" to be compatible with all GroupTypes?? I've noticed through doing this that every time a user links a Product to a Document at the `/link_product_form/xxxxx` URL, the document's type can change. Should this be limited to editing in the `/datadocument/edit/167497/` view only?? I may just be overthinking it here understanding that many Products could be made this way, though that is not likely the commonplace workflow. For now, I'm just thinking of making the `"Unidentified"` `DocumentType` in the migration and then adding it to be compatible with every `GroupType`.
